detox/e2e/support/server_api/post.js

```diff
@@ -16,6 +16,34 @@ import {getResponseFromError} from './common';
 // - return value defined by `@return`
 // ****************************************************************
 
+/**
+ * Create a new post in a channel. To create the post as a comment on another post, provide root_id.
+ * See https://api.mattermost.com/#tag/posts/paths/~1posts/post
+ * @param {string} option.channelId - The channel ID to post in
+ * @param {string} option.message - The message contents, can be formatted with Markdown
+ * @param {string} option.rootId - The post ID to comment on
+ * @param {Object} option.props - A general object property bag to attach to the post
+ * @param {Date} option.createAt - The date the post is created at
+ * @return {Object} returns {post} on success or {error, status} on error
+ */
+export const apiCreatePost = async ({channelId, message, rootId, props = {}, createAt = 0}) => {
+    try {
+        const payload = {
+            channel_id: channelId,
+            message,
+            parent_id: rootId,
+            root_id: rootId,
+            props,
+            create_at: createAt,
+        };
```

Comment:
I'll definitely explore this in a future PR. I think this would allow posting messages with files but not attaching files to post draft. I'll take a look at one of the gallery test cases and see if this is viable.